react.md

```diff
@@ -10,37 +10,286 @@ mainmenu: true
+
+<div className={styles.container}>
+  <article className={styles.main}>
+    Your main content
+  </article>
+  <aside className={styles.sidebar}>
+    Some sidebar content
+  </aside>
+</div>
+```
+
+Refactor it into a set of layout components, each with their own CSS module:
+```js
+import Layout from 'components/layout/Layout';
+
+<Layout>
+  <Layout.Main>
+    Your main content
+  </Layout.Main>
+  <Layout.Sidebar>
+    Some sidebar content
+  </Layout.Sidebar>
+</Layout>
+```
+
+#### ui
+UI components are non-form interactive elements like buttons, menus, modals, sliders, cards, icons, tooltips, and progress bars. If it doesn't fit in form or layout, it probably belongs here.
+
+If you're copying and pasting something like this into more than 1 view:
+```js
+import { Modal, Button } from 'some-react-ui-framework';
+import styles from 'globalStylesYouImportInEveryPage.module.css';
+
+<Modal
+  isOpen={this.isOpen}
+>
+  <h2 className={styles.modalTitle}>
+    Modal title
+  </h2>
+  <div className={styles.modalBody}>
+    Modal content
+  </div>
+  <footer className={styles.modalFooter}>
+    <Button
+      onClick={this.doSomething}
+      className={styles.modalButton}
+    >
+      Do A Thing
+    </Button>
+  </footer>
+</Modal>
+```
+
+Refactor it to a UI component with its own CSS module:
+```js
+import Modal from 'components/ui/Modal';
+
+<Modal
+  isOpen={this.isOpen}
+  title="Modal title"
+  actions={(
+    <Button onClick={this.doSomething}>Do A Thing</Button>
+  )}
+>
+  Modal content
+</Modal>
+```
+
+Note how our Modal's `actions` prop makes it easy for implementers to customize footer actions without adding boilerplate markup. Even if you're using a UI framework, you should still look for ways to reduce repetition and make it easier for other developers to implement your components.
+
+#### view-partials
+View partials are the non-interactive equivalent of form partials. This is where you put small re-usable templates for formatting & displaying data.
+
+If you're copying & pasting something like this into multiple pages:
+```js
+import styles from 'globalStylesYouImportInEveryPage.module.css';
+
+<address>
+  <div className={styles.line1}>
+    {user.address.streetNumber} {user.address.streetName}
+  </div>
+  {user.address.unit && (
+    <div className={styles.line2}>
+      Unit {user.address.unit}
+    </div>
+  )}
+  <div className={styles.cityProvincePostal}>
+    {user.address.city}, {user.address.province} {user.address.postalCode}
+  </div>
+</address>
+```
+
+Refactor it to a view partial with its own CSS module:
+```js
+import Address from 'components/view-partials/Address';
+
+<Address
+  address={user.address}
+/>
+```
+
+### Other files
+
+#### index.js
+This serves as the entry point for your application. This is typically where you'd import & mount your router and the page containers to be rendered by each route. CRA provides this file for you, and you'll find that it will grow as your application does.
```

Comment:
Is index.js where you hold your routes? It's a bit unclear - are we recommending not having a Routes.js file , and index.js for routes? Routes is one file that differs on nearly every project, when it shouldn't.

Reply:
I think it depends? We should probably expand in the routing section, but if you're using react-router 4, they recommend using a component-based approach instead of the "big array of routes" approach.

I can amend this to not mention routing so we can be more detailed in the later section.